docs/structured-streaming-programming-guide.md

```diff
@@ -3146,6 +3146,17 @@ See [Input Sources](#input-sources) and [Output Sinks](#output-sinks) sections f
       - After `coalesce`, the number of (reduced) tasks will be kept unless another shuffle happens.
   - `spark.sql.streaming.stateStore.providerClass`: To read the previous state of the query properly, the class of state store provider should be unchanged.
   - `spark.sql.streaming.multipleWatermarkPolicy`: Modification of this would lead inconsistent watermark value when query contains multiple watermarks, hence the policy should be unchanged.
+- Structured Streaming uses `global watermark` which might impact on query having multiple stateful operations.
```

Comment:
_a_ global watermark? it also shouldn't be code-formatted
impact on query -> impact a query.
How would it impact the query?

Reply:
Will fix on above two lines of review comment.

> How would it impact the query?

The answer is in below line - `Fail to answer above questions might lead to incorrect outputs - e.g. intermediate outputs being discarded.` Not sure the flow looks natural, or you might want to revise the format/flow of content.